### PR TITLE
[deutschebahn] Implemented filters for trains in timetable

### PR DESCRIPTION
**Pull request by soenkekueper**
To allow an easier configuration for each user, the timetable thing now supports custom filters.
So users may specify an filter expression and display only trains which they want to display
within the timetable.

### CODE REVIEW

File: bundles/org.openhab.binding.deutschebahn/README.md [hunk 1]
```diff
@@ -40,7 +40,27 @@ In addition you can configure if only arrivals, only departures or all trains sh
...
 | `trainFilter` |  | Yes | Selects the trains that will be displayed in the timetable. Either only arrivals, only departures or all trains can be displayed. |
+| `additionalFilter` | | No | Specifies additional filters for trains, that should be displayed within the timetable. |
 
+** Additional filter **
+If you only want to display certain trains within your timetable, you can specify an additional filter. This will be evaluated when loading trains,
+and only trains that matches the given filter will be contained within the timetable.
+
+To specify an advanced filter you can
+
+- specify an filter for the value of an given channel. Therefore you must specify the channel name (with channel group) and specify an compare value like this:
```
Comment (fwolter) [inline bundles/org.openhab.binding.deutschebahn/README.md:51]:
```suggestion
- specify a filter for the value of a given channel. Therefore you must specify the channel name (with channel group) and specify a compare value like this: 
```

File: bundles/org.openhab.binding.deutschebahn/README.md [hunk 2]
```diff
@@ -40,7 +40,27 @@ In addition you can configure if only arrivals, only departures or all trains sh
...
+** Additional filter **
+If you only want to display certain trains within your timetable, you can specify an additional filter. This will be evaluated when loading trains,
+and only trains that matches the given filter will be contained within the timetable.
+
+To specify an advanced filter you can
+
+- specify an filter for the value of an given channel. Therefore you must specify the channel name (with channel group) and specify an compare value like this: 
+`departure#line="RE60"` this will select all trains with line RE60
+- use regular expressions for expected channel values, for example:  `departure#line="RE.*"`, this will match all lines starting with "RE".
+- combine multiple statements as and conjunction by using `&`. If used both parts must match, for example: `departure#line="RE60" & trip#category="WFB"`
```
Comment (fwolter) [inline bundles/org.openhab.binding.deutschebahn/README.md:54]:
```suggestion
- combine multiple statements as "and" conjunction by using `&`. If used both parts must match, for example: `departure#line="RE60" & trip#category="WFB"`
```

File: bundles/org.openhab.binding.deutschebahn/README.md [hunk 3]
```diff
@@ -40,7 +40,27 @@ In addition you can configure if only arrivals, only departures or all trains sh
...
+- combine multiple statements as or disjunction by using `|`. If used one of the parts must match, for example: `departure#line="RE60" | departure#line="RE60"`
```
Comment (fwolter) [inline bundles/org.openhab.binding.deutschebahn/README.md:55]:
```suggestion
- combine multiple statements as "or" disjunction by using `|`. If used, one of the parts must match, for example: `departure#line="RE60" | departure#line="RE60"`
```

File: bundles/org.openhab.binding.deutschebahn/README.md [hunk 4]
```diff
@@ -40,7 +40,27 @@ In addition you can configure if only arrivals, only departures or all trains sh
...
+- use brackets to build more complex queries like `trip#category="RE" AND (departure#line="17" OR departure#line="57")`
+
+If an channel has multiple values, like the channels `arrival#planned-path` and `departure#planned-path` have a list of stations,
```
Comment (fwolter) [inline bundles/org.openhab.binding.deutschebahn/README.md:58]:
```suggestion
If a channel has multiple values, like the channels `arrival#planned-path` and `departure#planned-path` have a list of stations,
```

File: bundles/org.openhab.binding.deutschebahn/README.md [hunk 5]
```diff
@@ -40,7 +40,27 @@ In addition you can configure if only arrivals, only departures or all trains sh
...
+only one of the values must match the given expression. So you can specify an filter like `departure#planned-path="Hannover Hbf"`
```
Comment (fwolter) [inline bundles/org.openhab.binding.deutschebahn/README.md:59]:
```suggestion
only one of the values must match the given expression. So you can specify a filter like `departure#planned-path="Hannover Hbf"`
```

File: bundles/org.openhab.binding.deutschebahn/README.md [hunk 6]
```diff
@@ -40,7 +40,27 @@ In addition you can configure if only arrivals, only departures or all trains sh
...
+to easy display only trains that will go to Hannover Hbf.
```
Comment (fwolter) [inline bundles/org.openhab.binding.deutschebahn/README.md:60]:
```suggestion
to easily display only trains that will go to Hannover Hbf.
```

**Review by fwolter — CHANGES_REQUESTED** [review]

**Comment by soenkekueper** [comment]
Thanks for your review. I've applied the remarks now.

**Review by fwolter — APPROVED** [review]

**Pull request merged**